web3pi_proxy/core/rpc/node/endpoint_pool/endpoint_connection_pool.py

```diff
@@ -167,6 +170,12 @@ def __run_cleanup_thread(self) -> None:
     def __get_connection(self) -> EndpointConnection:
         return self.connections.get_nowait()
 
+    def new_connection(self) -> EndpointConnection:
+        """Internal function, do not call directly"""
```

Comment:
instead of naming it like that, please change the name to `__new_connection` or `_new_connection` to make it obvious that it's an internal function...